packages/graphql/tests/integration/directives/cypher/filtering/cypher-filtering-list.int.test.ts

```diff
@@ -0,0 +1,129 @@
+    const testHelper = new TestHelper();
+
+    afterEach(async () => {
+        await testHelper.close();
+    });
+
+    beforeEach(() => {
+        CustomType = testHelper.createUniqueType("CustomType");
+    });
+
+    test.each([
+        {
+            title: "String List cypher field: INCLUDES",
+            filter: `custom_cypher_list_INCLUDES: "a"`,
+            type: "String",
+            cypherStatement: "RETURN ['a', 'b', 'c'] as list",
+        },
+        {
+            title: "Int List cypher field: INCLUDES",
+            filter: `custom_cypher_list_INCLUDES: 2`,
+            type: "Int",
+            cypherStatement: "RETURN [1,2,3,4,5] as list",
+        },
+        {
+            title: "Float List cypher field: INCLUDES",
+            filter: `custom_cypher_list_INCLUDES: 2.0`,
+            type: "Float",
+            cypherStatement: "RETURN [1.0,2.0,3.0,4.0,5.0] as list",
+        },
+        {
+            title: "Point List cypher field: INCLUDES",
+            filter: `custom_cypher_list_INCLUDES: { latitude: 1, longitude: 2 }`,
+            type: "Point",
+            cypherStatement:
+                "RETURN [point({ latitude: 1, longitude: 2 }), point({ latitude: 3, longitude: 4 })] as list",
+        },
+        {
+            title: "CartesianPoint List cypher field: INCLUDES",
+            filter: `custom_cypher_list_INCLUDES: { x: 1, y: 2, z: 3 }`,
+            type: "CartesianPoint",
+            cypherStatement: "RETURN [point({ x: 1, y: 2, z: 3 }), point({ x: 3, y: 4, z: 5 })] as list",
+        },
+        {
+            title: "DateTime List cypher field: INCLUDES",
+            filter: `custom_cypher_list_INCLUDES: "2021-01-01T00:00:00Z"`,
+            type: "DateTime",
+            cypherStatement: "RETURN [datetime('2021-01-01T00:00:00Z'), datetime('2021-01-02T00:00:00Z')] as list",
+        },
+        {
+            title: "LocalDateTime List cypher field: INCLUDES",
+            filter: `custom_cypher_list_INCLUDES: "2021-01-01T00:00:00"`,
+            type: "LocalDateTime",
+            cypherStatement:
+                "RETURN [localdatetime('2021-01-01T00:00:00'), localdatetime('2021-01-02T00:00:00')] as list",
+        },
+        {
+            title: "Date List cypher field: INCLUDES",
+            filter: `custom_cypher_list_INCLUDES: "2021-01-01"`,
+            type: "Date",
+            cypherStatement: "RETURN [date('2021-01-01'), date('2021-01-02')] as list",
+        },
+        {
+            title: "Time List cypher field: INCLUDES",
+            filter: `custom_cypher_list_INCLUDES: "12:00:00"`,
+            type: "Time",
+            cypherStatement: "RETURN [time('12:00:00'), time('13:00:00')] as list",
+        },
+        {
+            title: "LocalTime List cypher field: INCLUDES",
+            filter: `custom_cypher_list_INCLUDES: "12:00:00"`,
+            type: "LocalTime",
+            cypherStatement: "RETURN [localtime('12:00:00'), localtime('13:00:00')] as list",
+        },
```

Comment:
With all of these tests, whilst we are testing that matching nodes are not filtered out, we are not really testing that non-matching nodes are filtered out.

Worthwhile creating the nodes with the custom Cypher return value as a property and having the custom Cypher simply return that property? Then you can create two nodes - one with with the property set and one without, and test that only the one with the property is returned.